File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun removeNthFromEnd(head: ListNode?, n: Int): ListNode? {
        var p: ListNode? = head
        var q: ListNode? = head
        for (i in 1..n){
            p = p?.next
        }
        if(p == null) return head?.next
        while(p?.next != null){
            p = p?.next
            q = q?.next
        }
        q?.next = q?.next?.next
        return head
    }
}

File: 19-remove-nth-node-from-end-of-list/19-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        p = head
        q = head
        for i in range(n):
            p = p.next
        if(p == None): return head.next
        while(p.next != None):
            p = p.next
            q = q.next
        q.next = q.next.next
        return head